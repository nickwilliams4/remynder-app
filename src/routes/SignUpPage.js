import React, { Component } from 'react'
import SignUpForm from '../SignUpForm/SignUpForm'
import { Section } from '../Utils/Utils'

export default class RegistrationPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }

  handleRegistrationSuccess = user => {
    const { history } = this.props
    history.push('/NotePage')
  }

  render() {
    return (
      <Section className='RegistrationPage'>
        <SignUpForm
          onRegistrationSuccess={this.handleRegistrationSuccess}
        />
      </Section>
    )
  }
}