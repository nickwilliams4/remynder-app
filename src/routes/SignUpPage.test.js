import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignUpPage from './SignUpPage'

describe(`SignUpPage component`, () => {
  const props = {
    className: 'test-class-name',
  }

  it('renders the SignUpPage', () => {
    const wrapper = shallow(<SignUpPage {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})