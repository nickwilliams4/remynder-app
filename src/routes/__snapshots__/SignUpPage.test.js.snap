// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage component renders the SignUpPage 1`] = `
<Section
  className="RegistrationPage"
>
  <SignUpForm
    onRegistrationSuccess={[Function]}
  />
</Section>
`;
